lib/temporal/client.rb

```diff
@@ -0,0 +1,382 @@
+require 'json'
+require 'securerandom'
+require 'socket'
+require 'temporal/errors'
+require 'temporal/converter'
+require 'temporal/interceptor/client'
+require 'temporal/interceptor/chain'
+require 'temporal/workflow/handle'
+require 'temporal/workflow/execution_info'
+require 'temporal/workflow/id_reuse_policy'
+require 'temporal/workflow/query_reject_condition'
+
+module Temporal
+  class Client
+    attr_reader :namespace, :identity
+
+    # TODO: More argument to follow for converters, codecs, etc
+    def initialize(connection, namespace, interceptors: [])
+      @connection = connection
+      @namespace = namespace
+      @identity = "#{Process.pid}@#{Socket.gethostname}"
+      @interceptor_chain = Interceptor::Chain.new(interceptors)
+      @converter = Converter.new
+    end
+
+    def start_workflow(
```

Comment:
@laertispappas it does make sense, however I don't think we should do it, because we won't be able to type it later on. RBS and Sorbet don't support variadic arguments. I however segregated the implementation of the Client and reduced it's API, so now it shouldn't look as repetitive

Reply:
That's a good point. I wonder if we could make an exception here for the wrapper classes? I assume letting the callers assign the input directly is not an option here right? Like:

```
opts = WorkflowClientInput.init.set_worflow_id(1).set_task_queue('task-queue').build
client.start_workflow(opts)
``` 

Although this exposes the clients with more inner details...